snpm_cp.m

```diff
@@ -782,6 +798,45 @@ function snpm_cp(CWD)
 %-Loop over planes (breaks out after first loop if bVolm)
 %-----------------------------------------------------------------------
 nP = [];
+
+if(UseRapidPT >= 1)
+    
+    params.N = size(X,1);
+    params.V = size(X,2);
+    params.nPerm = nPerm;
+    params.xdim = xdim; 
+    params.ydim = ydim; 
+    params.zdim = zdim; 
+    params.origin = ORIGIN; 
+    params.nGroup1 = size(find(iCond == 1),2); 
+    params.nGroup2 = params.N - params.nGroup1; 
+
+    datasetInfo = strcat(num2str(params.N),'_',num2str(params.nGroup1),'_',num2str(params.nGroup2));
+    runInfo = strcat(datasetInfo,'_',num2str(nPerm),'.mat');
+    save(strcat('params',runInfo),'params');
+
+    
+    fullpath = mfilename('fullpath');
+    % We only need the directory of snpm_cp so we remove the 7 chars
+    % associated to 'snpm_cp' in fullpath.
+    RapidPT_path = strcat(fullpath(1:end-7),'RapidPT_min');
+    addpath(RapidPT_path);
+
+    write = 0;
+    [outputs, timings] = TwoSampleRapidPT(X, nPerm, params.nGroup1, write, RapidPT_path);
+    MaxT = outputs.MaxT;
+    save(strcat('timings',runInfo),'timings');
+
+    % Save variables for snpm_pp
+    MaxT = [MaxT',-MaxT'];
+    % Calculate uncorrected p-vals
+    [~, SnPMucp, ~, ~] = ttest2(X(1:params.nGroup1, :), X(params.nGroup1+1:end, :), 0.05, 'both', 'unequal');
+    save('SnPMucp.mat','SnPMucp')
```

Comment:
Thanks for saving SnPMucp.mat, XYZ.mat, SnPMt.mat and updating SnPM.mat here. However, there is more bookkeeping that is done starting at the comment "write out lP+ and lP- images" below.  In particular, when RapidPT=1, for the images lP{+,-}, lP_FDR{+,-}, and lP_FWE{+,-}, the header files are written out above and then the img files are never created.  

I would prefer it if "end" for this "else" came higher up, so that the lP* images could be written out.  Let me know if you need a hand with this.

Also note that the progress bar is not appropriately closed out nor the timing reported, as is done in the code below starting at `if bWin, spm_progress_bar('Clear')...`.